src/main/webapp/gameDialogue.js

```diff
@@ -93,14 +98,60 @@ function modifyIfFinalStage() {
       });
 }
 
-//play button functions
-function playmusic() {
-  //TODO: change the play and pause to toggle on and off icons
-  if(document.getElementById('player-button').innerText == "PAUSE"){
-    document.getElementById('player-button').innerText = "PLAY";
+function getExp() {
+  fetch("/experience")
+  .then(response => response.text())
+  .then(expString => {
+    exp = parseInt(expString);
+    console.log("EXPERIENCE: " + exp);
+    showExp(exp);
+  });
+}
+
+function showExp(exp) {
+  const expContainer = document.getElementById('exp-container');
+  expContainer.innerText = "EXP:" + exp;
+}
+
+function changeExp() {
+  exp ++;
+  workPromoButtonSwitch();
+  showExp(exp);
+  const params = new URLSearchParams();
+  params.append('experience', exp);
+  fetch('/experience', {method: 'POST', body: params});
+}
+
+function workPromoButtonSwitch() {
+  if (exp == threshold) {
+    // Changes button to "TRY FOR PROMOTION"
+    const promoButton = document.getElementById('promotion-button');
+    promoButton.innerText = "TRY FOR PROMOTION";
+  } else if (exp == threshold + 1) {
+    // Changes button link to promotion.html
+    showPromoButton(true, null);
```

Comment:
Add comment explaining the null argument.

Reply:
+1